sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/feature/FeatureFlagState.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Is there any difference between `State` and `FeatureFlagState`? Is it possible that we make `State` as base class and `FeatureFlagState extend State`?
 https://github.com/Azure/azure-sdk-for-java/blob/feature/spring-boot-3/sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/State.java

Reply:
Pre this PR there was only one `State`. The idea was that `State` had the Configuration Setting that was the watch key, which contains the etag info and we would just go through the list to see if any of them changed. This also worked for feature flags.

Feature Flags now refresh by page, so we need to track the etags by the page, which is part of the `MatchConditions`, which is inside of a `SettingSelector`.  If you look at the confusingly named `FeatureFlags` object which is new it contains this `SettingSelector`. This is how we now have to track changes for all feature flags.

I'm not a huge fan of this setup, there are a number of reasons why I have this up as a draft. This part will should also work when we add page monitoring for ConfigurationSettings, so we may want to update the names to reflect this.